assets/js/components/Notice/Icon.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Null from '../Null';
+import CheckFill from '../../../svg/icons/check-fill.svg';
+import WarningSVG from '../../../svg/icons/warning.svg';
+import StarFill from '../../../svg/icons/star-fill.svg';
+
+export const TYPE_NEW = 'new';
+export const TYPE_SUCCESS = 'success';
+export const TYPE_WARNING = 'warning';
+export const TYPE_INFO = 'info';
+export const TYPE_ERROR = 'error';
```

Comment:
These are how the icons are keyed, but are really specific to the `type` prop for the component so it should probably be an export from the index (although that probably isn't possible due to cyclical imports) in which case it could be a separate. Also, for things like this, let's use the enum style for defining them all on a single export.

Maybe we could define it as `Notice.TYPES` on the component?